uaclient/apt.py

```diff
@@ -131,15 +131,8 @@ def add_apt_auth_conf_entry(repo_url, login, password):
     util.write_file(apt_auth_file, '\n'.join(new_lines), mode=0o600)
 
 
-def remove_auth_apt_repo(repo_filename, repo_url, keyring_file=None,
-                         fingerprint=None):
-    """Remove an authenticated apt repo and credentials to the system"""
-    logging.info('Removing authenticated apt repo: %s', repo_url)
-    util.del_file(repo_filename)
-    if keyring_file:
-        util.del_file(keyring_file)
-    elif fingerprint:
-        util.subp(['apt-key', 'del', fingerprint], capture=True)
+def remove_repo_from_apt_auth_file(repo_url):
```

Comment:
Thanks for adding the unit test for esm, is it work a minimal unit test for remove_repo_from_apt_auth_file and/or remove_auth_repo too?

Reply:
Yep, will add.

Reply:
@blackboxsw I've just pushed up one half of these tests because writing them raised a couple of questions about the implementation of `remove_auth_apt_repo`.

Firstly, it will only use the `fingerprint` for deletion if `keyring_file` is not given; should we unconditionally perform both actions if both are given?

Secondly, the `fingerprint` parameter doesn't appear to be passed in by any consumer. Do we actually want to keep it, or can we simplify the codebase a little?

(Whatever the answer to each of these questions is, I think we can open an issue to address it; I've already written tests for the existing behaviour, so we won't save any time by addressing it in this PR.)

Reply:
@OddBloke I think the way uaclient is headed, we ultimately want to drop static keyrings from uaclient if we can. The intent of the new client talking to ua-contracts backend was to primarily support the fingerprint or short key that the backend provides instead of these static keys.  I don't think we should perform the install operation on both keys. They probably should be mutually exclusive or at least  prefer the backend advertised key if present.

Reply:
+1 on separate PRs

Reply:
I just opened #394 to cover the removal of the parameter, and #395 to cover unconditional deletion.